### ISSUE TITLE
quarkus-kubernetes and quarkus-maven-plugin:native-image - Build failed due to errors

### ISSUE THREAD

**Issue by rsvoboda** [quarkusio/quarkus#7185]
When I use quarkus-kubernetes extension (Quarkus master based) and quarkus-maven-plugin:native-image (deprecated approach, but still available) based native profile I get build failure.

This is the use case when user updates the version of Quarkus in already existing project. When creating new project without native-image goal everything works.

I did some investigation and the main diff is in the resolved path for https://github.com/quarkusio/quarkus/blob/master/extensions/kubernetes/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java#L236

`artifactPath` for old quarkus-maven-plugin:native-image is `/fooOld-1.0.0-SNAPSHOT-runner.jar`. 
For the current way it is `/Users/rsvoboda/Downloads/fooBar/target/classes/fooBar-1.0.0-SNAPSHOT-runner.jar`.

So the path for quarkus-maven-plugin:native-image is cropped and not an absolute path.

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:native-image (default) on project 003-quarkus-many-extensions: Failed to generate native image: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.kubernetes.deployment.KubernetesProcessor#build threw an exception: java.lang.UnsupportedOperationException
[ERROR] 	at com.sun.nio.zipfs.ZipPath.toFile(ZipPath.java:597)
[ERROR] 	at io.quarkus.kubernetes.deployment.KubernetesProcessor.createProject(KubernetesProcessor.java:236)
[ERROR] 	at io.quarkus.kubernetes.deployment.KubernetesProcessor.build(KubernetesProcessor.java:133)
```

Steps to reproduce:
```
mvn io.quarkus:quarkus-maven-plugin:0.26.1:create \
    -DprojectGroupId=io.quarkus.qe \
    -DprojectArtifactId=fooOld \
    -DprojectVersion=1.0.0-SNAPSHOT \
    -DplatformArtifactId=quarkus-bom \
    -DclassName="io.quarkus.qe.MyResource" \
    -Dextensions="kubernetes"
cd fooOld
sed -i "" "s/0.26.1/999-SNAPSHOT/g" pom.xml
mvn clean verify -Dnative -DskipTests -DskipITs
```

**Comment by geoand** [comment]
Let's wait for the major changes in #5496 to land and then look at this one

**Comment by gsmet** [comment]
I don't think it's worth spending time on that one.

People using the latest versions won't have this issue and we didn't have any report.

**Comment by rsvoboda** [comment]
Tried it with merged changes from #5496 and still get java.lang.UnsupportedOperationException

We can leave it as open and wait if people come and comment on this issue. Otherwise Stale bot will close it.

**Comment by geoand** [comment]
Sure, we can leave it open for the time being although I doubt anyone will pick it up anytime soon unless someone complains :)

**Comment by geoand** [comment]
This has since been fixed in #8322
